gtsam/discrete/DecisionTree.h

```diff
@@ -234,11 +233,13 @@ namespace gtsam {
 
     /**
      * @brief Visit all leaves in depth-first fashion.
-     * 
-     * @param f side-effect taking a value.
-     * 
-     * @note Due to pruning, leaves might not exhaust choices.
-     * 
+     *
```

Comment:
I think these functions passed in  should now also take an argument for the number of assignments captured in a particular leaf.

Reply:
I added `visitLeaf` as an alternative to that. Should I remove that and update the rest to do this (will be API breaking)?

Reply:
Shoot, I merged too soon! I just noticed this new visitLeaf, which in this PR seems identical??

Reply:
And the examples are wrong I think... We can talk about it in our meeting.

Reply:
Okay! Though I don't want to spend more than 5 minutes on this since it is secondary to the workshop paper.